bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/setup/ThingSetupManager.java

```diff
@@ -260,6 +260,12 @@ public void enableChannel(ChannelUID channelUID) {
                     }
                     item.addTags(channelType.getTags());
                     item.setCategory(channelType.getCategory());
+                    if (channelType.getState().getPattern() != null) {
+                        String label = channelType.getLabel() + " [" + channelType.getState().getPattern() + "]";
```

Comment:
> This is currently not the case - as you can see in the demo.items, the formatting is still part of the label and it is required by e.g. the Classic UI. Btw, this PR is all about fixing the code to make it work with the Classic UI, which is completely broken at the moment as it has not been considered in the past refactorings. And I expect the same for all other REST-based clients besides Paper UI.

Well the Paper UI is so far the only client, which uses the new concepts like state descriptions or the category. But all other UIs render based on sitemaps, right? So why not change the implementation of the getLabel() method in the ItemUIRegistry? If we will change this here, the Paper UI will render it wrong, as the UI will not format the label. Questions to me is, how should other UIs render it in the future (independently how it was done in the past) and should a pattern be specified as part of the label? If we want to avoid this in the future, we should better workaround it in the Classic UI, or in the sitemap for the moment.

> I would therefore suggest to do it at the moment this way, until we might have implemented https://www.eclipse.org/forums/index.php/t/1027209/.

As far as I understand this forum post, it is about adding an additional field "formattedValue" in the ItemBean of the Item REST resource. We can implement it easily, but I´m not sure if this helps for the Classic UI.